### PR TITLE
feat: add GoogleAds connector

### LINKED ISSUE #219: feature: Google Analytics Connector

### PR DESCRIPTION
**Pull request by remitache**
This PR adds support for Google Ads connector
Fixes #219

### CODE REVIEW

**Comment by psrebniak** [comment]
Thanks @remitache for raising the PR, appreciate your contribution

Few adjustments required:
- run `npm run projen` to regenerate autogenerated files and run linter on the code, most probably if you fix errors here, next point will not be valid anymore
- tests are failing - please take a look: https://github.com/cdklabs/cdk-appflow/actions/runs/8975852042/job/24651396941?pr=200
- PR has to be linked with issue

Tomorrow I'll add integration test and test it using real connection.

File: src/googleads/profile.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        super(scope, id, props, GoogleAdsConnectorType.instance);
+    }
+
+
+    protected buildConnectorProfileProperties(
+        props: ConnectorProfileProps,
+      ): CfnConnectorProfile.ConnectorProfilePropertiesProperty {
+        const properties = (props as GoogleAdsConnectorProfileProps);
+        return {
+          customConnector: {
```
Comment (psrebniak) [inline src/googleads/profile.ts:99]: `customConnector` should also contain following property: 
```
            profileProperties: {
              developerToken: properties.developerToken.unsafeUnwrap(),
              apiVersion: properties.apiVersion,
              managerID: properties.managerID?.unsafeUnwrap() as any /* can be undefined */,
            },
```

File: src/googleads/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/*
```
Comment (psrebniak) [inline src/googleads/index.ts:1]: need to fix linter - please run `npm run projen` and fix errors once you address remaining comments

File: test/googleads/source.test.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+            AuthenticationType: 'OAUTH2',
+            Oauth2: {
+              AccessToken: 'accessToken',
+              ClientId: 'clientId',
+              ClientSecret: 'clientSecret',
+              RefreshToken: 'refreshToken',
+            },
+          },
+        },
+        ConnectorProfileProperties: {
```
Comment (psrebniak) [inline test/googleads/source.test.ts:159]: should have few more props here - please check my previous comments

**Review by psrebniak — CHANGES_REQUESTED** [review]
Two adjustments needed, build is failing due to invalid imports / eslint / tests. 
Once this is solved and PR is attached to issue, we can merge it - really good piece of code!

**Comment by psrebniak** [comment]
@remitache build still failing - are you sure you added every file to commit? Seems like some of autogenerated files were not added

**Comment by remitache** [comment]
@psrebniak Yes I pushed all the file but merging still fail. I am not sure why

**Comment by psrebniak** [comment]
Can you please make sure you're in root directory of project and run 
```sh
yarn run projen
git add .
git commit -m 'autogenerated files'
git push
```
? 
I tried it now and see two files generated. 
If this does not work, then I'll raise PR against your branch to fix CI

**Comment by psrebniak** [comment]
@remitache ok, apparently `yarn run projen` is not enough, can you do the same with `yarn run build` please?

**Comment by psrebniak** [comment]
@remitache now it works, one last step required: please create an issue and link your PR for visibility, tomorrow I'll do last check and will merge it. 
Really good work!

**Comment by remitache** [comment]
@psrebniak I am not able to link an issue with the PR. Just created an empty issue for now.

**Review by psrebniak — APPROVED** [review]
looks good, thanks for your contribution @remitache

**Pull request merged**